backend/Api/Projects/ProjectModels.cs

```diff
@@ -15,24 +14,36 @@ public record EngagementReadModel(
     [property: Required] EngagementState BookingType,
     [property: Required] bool IsBillable);
 
-public record EngagementWriteModel(EngagementState BookingType,
-    bool IsBillable, string ProjectName, string CustomerName);
+public record EngagementWriteModel(
+    EngagementState BookingType,
+    bool IsBillable,
+    string ProjectName,
+    string CustomerName);
 
 public record ProjectWithCustomerModel(
     [property: Required] string ProjectName,
     [property: Required] string CustomerName,
     [property: Required] EngagementState BookingType,
     [property: Required] bool IsBillable,
-    [property: Required] int ProjectId);
-
-public record UpdateProjectWriteModel(int EngagementId, EngagementState ProjectState, int StartYear, int StartWeek,
+    [property: Required] int ProjectId)
```

Comment:
Hva gjør egentlig `[property: Required]`?

Reply:
Ah! Det kan du fikse når du legger til swagger i program.cs
```csharp
services.AddSwaggerGen(options =>
{
	options.SupportNonNullableReferenceTypes();
```

Reply:
Siden det er redundant å spesifisere på records, under the hood så gjør de felter som `public record MyRecord(string Id)` til `public required string Id {public get; public init;}`